src/index.js

```diff
@@ -74,3 +74,39 @@ deleteBtn.addEventListener('click', () => {
 });
 
 renderTasks();
+// readTask();
+updateTask();
+
+clearCompletedBtn.addEventListener("click", clearCompleted);
+
+function clearCompleted() {
+  const tasks = readTask();
+  const updatedTasks = tasks.filter(task => !task.completed);
+  updatedTasks.forEach((task, index) => {
+    task.id = index; 
+  });
+  updateTask(updatedTasks); 
+  renderTasks(); 
+}
+
+// Add event listener to each checkbox
+todoList.addEventListener("change", function(event) {
+  if (event.target.matches("input[type='checkbox']")) {
+    const checkbox = event.target;
+    const listItem = checkbox.closest("li"); 
+    const index = checkbox.dataset.index;
+    const tasks = readTask();
+
+    tasks[index].completed = checkbox.checked;
+    updateTask(tasks); 
+
+    if (checkbox.checked) {
+      listItem.classList.add("completed"); 
+    } else {
+      listItem.classList.remove("completed");
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=the%20page%20reloads%20(-,which%20means%20that%20any%20changes%20made%20to%20the%20list%20should%20be%20preserved,-).), Please toggle the completed to true or false at the local storage with the change of the checkbox.